pkg/machines/components/vmOverviewTabLibvirt.jsx

```diff
@@ -120,67 +159,130 @@ class VmOverviewTabLibvirt extends React.Component {
+                    disabled={config.provider.name !== "LibvirtDBus"}
+                    checked={config.provider.name == "LibvirtDBus" ? vm.autostart : vm.autostart != "disable" }
+                    onChange={this.onAutostartChanged} />
+                {_("Run when host boots")}
+            </label>
+        );
+        const bootOrder = (
+            <div>
+                <button className="link-button" disabled={config.provider.name !== "LibvirtDBus"} id={`${idPrefix}-boot-order`} onClick={this.openBootOrder}>
+                    {getBootOrder(vm)}
                 </button>
-            );
-        }
+                { vm.state === "running" && bootOrderChanged() && <WarningInactive iconId="boot-order-tooltip" tooltipId="tip-boot-order" /> }
+            </div>
+        );
+        const memoryLink = (
+            <div>
+                <button className="link-button" disabled={config.provider.name !== "LibvirtDBus"} id={`${idPrefix}-memory-count`} onClick={this.openMemory}>
+                    {cockpit.format_bytes(vm.currentMemory * 1024)}
+                </button>
+            </div>
+        );
         const vcpuLink = (
-            <>
-                <button className="link-button" id={`${vmId(vm.name)}-vcpus-count`} onClick={this.openVcpu}>{vm.vcpus.count}</button>
+            <div>
+                <button className="link-button" id={`${idPrefix}-vcpus-count`} onClick={this.openVcpu}>{vm.vcpus.count}</button>
                 { vm.state === "running" && vcpusChanged && <WarningInactive iconId="vcpus-tooltip" tooltipId="tip-vcpus" /> }
-            </>
+            </div>
         );
 
-        const items = [
-            { title: commonTitles.MEMORY, value: memoryLink, idPostfix: 'memory' },
-            { title: _("Emulated Machine"), value: vm.emulatedMachine, idPostfix: 'emulatedmachine' },
-            { title: commonTitles.CPUS, value: vcpuLink, idPostfix: 'vcpus' },
-            { title: _("Boot Order"), value: bootOrder, idPostfix: 'bootorder' },
-            { title: _("CPU Type"), value: vm.cpu.model, idPostfix: 'cputype' },
-            { title: _("Autostart"), value: autostart, idPostfix: 'autostart' },
-        ];
+        let firmwareLinkWrapper;
+        // <os firmware=[bios/efi]' settings is available only for libvirt version >= 5.2. Before that version it silently ignores this attribute in the XML
+        if (config.provider.name == 'LibvirtDBus' && this.state.loaderElems && libvirtVersion >= 5002000) {
+            const hasInstallPhase = vm.metadata.hasInstallPhase;
+            const labelForFirmware = labelForFirmwarePath(vm.loader, vm.arch);
+            let currentFirmware;
+            if (vm.firmware || labelForFirmware == "efi")
+                currentFirmware = "UEFI";
+            else if (labelForFirmware == "custom")
+                currentFirmware = cockpit.format(_("Custom firmware: $0"), vm.loader);
+            else
+                currentFirmware = "BIOS";
```

Comment:
This needs to handle the `unknown` value as well.

Reply:
Added condition for that.